rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+  aggregated by unique users.
+
+#### How usage grows or shrinks over time?
+
+PostHog allows to create [Trends](https://posthog.com/manual/trends) basing on DAU. It can be used to show how usage
+changes in a given period of time.
+
+## Events
+
+### `connect.start`
+
+Starting the app.
+
+Event properties:
+
+- `os`: one of `mac`/`windows`/`linux`
+- `connect_version`: string
+
+### `connect.cluster.add`
+
+Successful adding a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string - name of the added cluster
+
+### `connect.cluster.login`
+
+Successful login to a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `connector_type`: string
+
+### `connect.workspace.change`
+
+Changing a workspace.
+
+Event properties:
+
+- `target_root_cluster_name`: string
+
+### `connect.leafClusterPicker.change`
+
+Changing a leaf cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `target_leaf_cluster_name`: string
+
+### `connect.tab.open`
+
+Opening a tab.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `tab_kind`: one
+  of `resource_list`/`terminal_ssh`/`terminal_kube`/`terminal_shell`/`db_proxy`/`create_access_request`/`review_access_request`
+
+### `connect.connectionsPicker.item.select`
+
+Selecting an item from the connections picker.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `connection_kind`: one of `ssh`/`kube`/`db`
+
+### `connect.commandBar.command.run`
+
+Running a command from command bar.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+
+### `connect.commandBbar.autocomplete.select`
+
+Selecting autocomplete item from the command bar suggestions.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string
+- `suggestion`: one of `ssh_login`/`ssh_server`/`database`
+
+### `connect.resourceTable.search`
+
+Performing a resource search.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string
+- `resource_kind`: one of `ssh`/`kube`/`db`
+- `is_advanced`: boolean
+
+### `connect.accessRequests.create`
+
+Creating an access request.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `requested_resources_count`: number
+
+### `connect.accessRequests.review`
+
+Reviewing an access request.
+
+Event properties:
+
+- `root_cluster_name`: string
+
+### `connect.accessRequests.assumeRole`
+
+Assuming a requested role.
+
+Event properties:
+
+- `root_cluster_name`: string
+
+### `connect.fileTransfer.run`
+
+Running file transfer.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `leaf_cluster_name`: string (optional)
+- `direction`: one of `upload`/`download`
+
+### `connect.dbProxy.runCli`
+
+Clicking on _Run_ in DB proxy connection (to measure how many people use CLI to interact with DB).
```

Comment:
I'd say it'd be more to measure how many people use Connect to interact with the db through a CLI client. You can still create a db proxy and connect through a CLI client outside of Connect.

On that note, we should have another event for just creating a db proxy to measure people using Connect to interact with the proxy vs using other means (GUI clients or CLI clients outside of Connect).